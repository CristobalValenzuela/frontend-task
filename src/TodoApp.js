import React from "react";
import AddTodo from "./components/AddTodo";
import TodoList from "./components/TodoList";
import VisibilityFilters from "./components/VisibilityFilters";
import "./styles.css";
import {
  Card,
  CardContent,
  CardHeader,
  Container,
  CardActions,
} from "@material-ui/core";
import TaskDialog from "./components/TaskDialog";

export default function TodoApp() {
  const [open, setOpen] = React.useState(false);
  const [todo, setTodo] = React.useState();

  const handleSetTodo = (todo) => {
    setTodo(todo);
    setOpen(true);
  }

  const handleClickOpen = () => {
    setTodo(undefined);
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div>
      <Container maxWidth="md">
        <Card>
          <CardHeader
            title="Lista de Tareas"
            action={<AddTodo openDialog={handleClickOpen} />}
          />
          <CardContent>
            <TodoList setTodo={handleSetTodo}/>
          </CardContent>
          <CardActions disableSpacing>
            <VisibilityFilters />
          </CardActions>
        </Card>
      </Container>
      <TaskDialog handleClose={handleClose} open={open} todo={todo}/>
    </div>
  );
}
