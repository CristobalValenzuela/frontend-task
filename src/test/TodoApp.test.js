import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import TodoApp from "../TodoApp";

describe("Probamos el componente <TodoApp />", () => {
  let wrapper = shallow(<TodoApp />);

  beforeEach(() => {
    wrapper = shallow(<TodoApp />);
  });
  test("deber√≠a mostrar <TodoApp /> correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
