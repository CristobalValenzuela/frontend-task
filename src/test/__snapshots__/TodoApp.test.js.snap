// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probamos el componente <TodoApp /> deber√≠a mostrar <TodoApp /> correctamente  1`] = `
<div>
  <WithStyles(ForwardRef(Container))
    maxWidth="md"
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(CardHeader))
        action={
          <AddTodo
            openDialog={[Function]}
          />
        }
        title="Lista de Tareas"
      />
      <WithStyles(ForwardRef(CardContent))>
        <Connect(TodoList)
          setTodo={[Function]}
        />
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))
        disableSpacing={true}
      >
        <Connect(VisibilityFilters) />
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Container))>
  <Connect(TaskDialog)
    handleClose={[Function]}
    open={false}
  />
</div>
`;
